### PR TITLE
feat: Use Snuba for Group.from_event_id

### PR DESCRIPTION
**Pull request by lynnagara**
This is the first part of https://github.com/getsentry/sentry/pull/13905,
which updates the group model to use events from Snuba instead of
Postgres. Splitting into smaller changes to make it easier to review and
merge safely. This PR updates the from_event_id function.

### CODE REVIEW

File: src/sentry/models/group.py [hunk 1]
```diff
@@ -158,24 +158,13 @@ def from_event_id(self, project, event_id):
...
-        # Look up event_id in both Event and EventMapping,
-        # and bail when it matches one of them, prioritizing
-        # Event since it contains more history.
-        for model in Event, EventMapping:
-            try:
-                group_id = model.objects.filter(
-                    project_id=project.id,
-                    event_id=event_id,
-                ).values_list('group_id', flat=True)[0]
+        event = SnubaEvent.objects.from_event_id(event_id, project.id)
```
Comment (fpacifici) [inline src/sentry/models/group.py]: Discussed in https://github.com/getsentry/sentry/pull/13905/files/3d3aa8be9807b50b6548c0b1b5a6f5b1027b5f4e#r301352481

this is not used in the ingestion path, so it should be safe.

File: src/sentry/models/group.py [hunk 2]
```diff
@@ -158,24 +158,13 @@ def from_event_id(self, project, event_id):
...
         a Group for which it is found.
         """
-        from sentry.models import EventMapping, Event
+        from sentry.models import SnubaEvent
         group_id = None
 
-        # Look up event_id in both Event and EventMapping,
-        # and bail when it matches one of them, prioritizing
-        # Event since it contains more history.
-        for model in Event, EventMapping:
```
Comment (fpacifici) [inline src/sentry/models/group.py]: Discussed in https://github.com/getsentry/sentry/pull/13905/files/3d3aa8be9807b50b6548c0b1b5a6f5b1027b5f4e#r301352764

removing EventMapping dependency should be safe.

File: src/sentry/models/group.py [hunk 3]
```diff
@@ -158,24 +158,13 @@ def from_event_id(self, project, event_id):
...
 
-                # It's possible that group_id is NULL
-                if group_id is not None:
-                    break
-            except IndexError:
-                pass
+        if event:
+            group_id = event.group_id
 
         if group_id is None:
```
Comment (fpacifici) [inline src/sentry/models/group.py]: I am pretty confident issueless events the group id will be None, but could you please double check that ?

**Review by fpacifici — APPROVED** [review]

File: src/sentry/models/group.py (hunk 3 above)
Comment (lynnagara) [reply src/sentry/models/group.py]: Yes, issueless events still seems to work ok

**Pull request merged**